README.md

```diff
@@ -1 +1,80 @@
-# strong-params-basics
+# Strong Params Basics
+
+## What are Strong Params?
+
+To understand the goal of strong params, let's pretend that you run a pharmacy. What would happen if you let all prescription orders come through without checking for: valid prescriptions, driver licenses, etc.? (Spoiler alert: you'd probably end up in jail). It would be insane to run a pharmacy without verifying that orders were legitimate. In the same way Rails application (starting in Rails 4+) wanted to shore up some security vulnerabilities and now require that you whitelist the parameters that are permitted when you're sending form data to the database.
+
+
+## Setup
+
+To prevent confusion, in previous lessons I manually turned off the strong parameter requirement, to start this lesson I've gone with the Rails 4+ default that requires you to use strong parameters. If for any reason you ever need to turn them off you can add the line `config.action_controller.permit_all_parameters = true` to the `config/application.rb` file.
+
+
+## Code Implementation
+
+Let's begin by running the rails server and navigating to `localhost:3000/posts/new`, once there fill out the form and click `submit`. You'll see we get the following `ForbiddenAttributesError`:
+
+![ForbiddenAttributesError](https://s3.amazonaws.com/flatiron-bucket/readme-lessons/ForbiddenAttributesError.png)
+
+What this means is that Rails needs to be told what parameters are allowed to be submitted through the form to the database. Remembering back to our pharmacy example, this is essentially like a pharmacist not giving out a prescription out unless they have:
+
+1. A valid prescription from a doctor
+
+2. The customer requesting the item matches the name on the prescription
+
+The same error would occur if you were trying to update a record, so how do we fix this? Let's update the `create` and `update` methods to look like the code below:
+
+```ruby
+# app/controllers/posts_controller.rb
+
+def create
+  @post = Post.new(params.require(:post).permit(:title, :description))
```

Comment:
explain the diff between `#require` (must have this!) and `#permit` (is allowed, but will still pass if not available)